supervisor/container.go

```diff
@@ -30,9 +30,16 @@ type Container struct {
 	ownerPod *HyperPod
 }
 
-func (c *Container) run(p *Process) {
+func (c *Container) run(p *Process) error {
+	// Receives the early result of the attempt to start the container,
+	// which ensures that we do not return before EventContainerStart
+	// is emitted.
```

Comment:
I'll revisit this code in that case, as this might require a more elaborate change to be a proper fix (and I'd prefer to fix it properly).